### ISSUE TITLE
Shift-click in inventory not working

### ISSUE THREAD

**Issue by Mathis-Z** [minetest/minetest#11791]
OS: MacOS 11.5.2
Minetest: 5.5.0-dev-206e131-dirty
IrrlichtMt: 1.9

Shift-clicking on items in the inventory does not work.
https://github.com/minetest/minetest/blob/206e131854392ed2d39b3456f7a1b5a54bd1bff9/src/gui/guiFormSpecMenu.cpp#L4286-L4294

event.MouseInput.Shift will always be false. I changed the code so it would set a boolean "shift_pressed" whenever there was an event with event.KeyInput.Key==KEY_SHIFT && event.KeyInput.PressedDown. Then I replaced !event.MouseInput.Shift with "shift_pressed". That worked but I did not test it very much yet.
Here is what I changed the code to:

At the beginning of GUIFormSpecMenu::OnEvent:
```
bool shift_pressed = false;
bool GUIFormSpecMenu::OnEvent(const SEvent& event)
{
	if (event.KeyInput.Key==KEY_SHIFT){
		if (event.KeyInput.PressedDown){
			shift_pressed = true;
		}else{
			shift_pressed = false;
		}
	}
      // more code...
    
```
And then I replaced !event.MouseInput.Shift:
```
if (!shift_pressed) {
	// no shift: select item
	m_selected_amount = count;
	m_selected_dragging = button != BET_WHEEL_DOWN;
	m_auto_place = false;
	} else {
		// shift pressed: move item, right click moves 1
		shift_move_amount = button == BET_RIGHT ? 1 : count;
	}
```

**Comment by sfan5** [comment]
duplicate of #9308

We've had a very similar fix (e18b6c5a21880125284d2a1e9dfb5471d5bc2bec) before, which was then reverted because it also broke other things.

> Minetest: most recent
> Irrlicht: 1.9

Irrlicht 1.9 is not a released version and neither does Minetest 5.4.1 (the latest release) support it.
Please re-test by compiling 5.5.0-dev from source, which relies on [IrrlichtMt](https://github.com/minetest/irrlicht) instead, to confirm whether this bug **really** still exists.

**Comment by Dieter44** [comment]
Sorry, I did not express this clearly: I compiled the code using the instructions from the minetest README.md. The version that is displayed on top of my minetest window is 5.5.0-dev-206e131-dirty
Actually I am not using Irrlicht but IrrlichtMt. I got the version number of IrrlichtMt from the README.md in lib/irrlichtmt
I edited the original post.

**Comment by sfan5** [comment]
I see.
It would be better to fix shift input directly in IrrlichtMt. If you want you can take a look at https://github.com/minetest/irrlicht/blob/master/source/Irrlicht/CIrrDeviceOSX.mm#L1011, a variable called `IsShiftDown` apparently still exists.

**Comment by TMcSquared** [comment]
@Zughy I can confirm that this used to be a bug. However in 5.5, It is no longer an issue for me.

**Comment by SmallJoker** [comment]
Closing until someone has this problem again.

**Comment by varttist** [comment]
@SmallJoker I seem to have encountered this problem

I noticed that Shift + LMB doesn't work for me, but Shift + RMB (or the mouse wheel) does.

OC:
```
Operating System: Arch Linux    
Kernel: Linux 7.0.5-arch1-1
```
Luanti:
- I noticed it on 5.15 (yes, I know it's old, but Luanti is updated with a delay in the Arch Linux package, and then I simply didn't update the entire system)
- I checked it on the new version 5.17-dev (compiled from source commit 65f2bfef9), the error is identical. Version:
```
Luanti 5.17.0-dev-65f2bfef9 (Linux)
Using LuaJIT 2.1.1774896198
Built by GCC 16.1
Running on Linux/7.0.5 x86_64
BUILD_TYPE=Release
RUN_IN_PLACE=1
USE_CURL=1
USE_GETTEXT=1
USE_SOUND=1
STATIC_SHAREDIR="."
STATIC_LOCALEDIR="locale"
```
I also tried to build version 5.5, and CMake initially warned me that it was missing the IrrlichtMt library. I cloned it, but the compilation failed with an error. I never tried compiling the older version again.

I also checked whether any application intercepts the Shift+LMB combination, but using the `xev` utility it is clear that the combination reaches the window

I plan to update the system later, maybe this will fix the problem
